### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by StijnDeJong**
Add logging that logs all keyword calls and data that goes around in the library. Errors raised by this library also get logged.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -28,13 +30,15 @@ def chatbot():
...
     pytest.main(["utest/"])
 
 if __name__ == "__main__":
-    real_test_data_generator()
-    chatbot()
-    robot_tests()
-    pytests()
+    setup_logging(enabled=True, for_tests=False, console_logging=False ,file_logging=True)
+    
+    # real_test_data_generator()
```
Comment (xfoggi) [inline src/main.py:41]: remove ?

**Review by xfoggi — CHANGES_REQUESTED** [review]

File: src/RobotFrameworkAI/__init__.py [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+from .logger.logger_config import setup_logging
 
+@library
 class RobotFrameworkAI(RealTestDataGenerator, Chatbot):
     """
     """
-    
+    @keyword
+    @wraps(setup_logging)
+    def setup_logging(self, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (StijnDeJong) [reply src/RobotFrameworkAI/__init__.py:15]: The wrapper automatically gets the documentation of the original method and adds.
image

File: src/RobotFrameworkAI/ai_interface/AI_Interface.py [hunk 3]
```diff
@@ -23,11 +27,21 @@ def __init__(self) -> None:
     
     def send_prompt(self, prompt):
         ai_model = prompt.config.ai_model
-        if ai_model not in self.ai_models:    
-            raise ValueError(f"Invalid ai_model: '{ai_model}'. Valid models are: {', '.join(self.ai_models)}")
+        try:
```
[the remark this answers is not in the document]
Comment (StijnDeJong) [reply src/RobotFrameworkAI/ai_interface/AI_Interface.py:30]: Fixed

[the remark this answers is not in the document]
Comment (StijnDeJong) [reply src/RobotFrameworkAI/ai_interface/AI_Interface.py:30]: Fixed

File: src/main.py (hunk 1 above)
Comment (StijnDeJong) [reply src/main.py:41]: I use the main.py as a way of showcasing the functionality. This helps to show how each module can be used and run the tests. Also is used to quickly test stuff

File: src/RobotFrameworkAI/modules/Module.py [hunk 4]
```diff
@@ -101,17 +105,21 @@ def get_default_values_for_common_arguments(
     def validate_common_input_arguments(self, max_tokens:int, temperature:float, top_p:float, frequency_penalty:float, presence_penalty:float):
         error_messages = []
         if not self.is_valid_max_tokens(max_tokens):
-            error_messages.append(f"Invalid value '{max_tokens}' for 'max_tokens'. Value must be greater than 0 and less than or equal to 4096.")
+            error_messages.append(f"Invalid value `{max_tokens}` for `max_tokens`. Value must be greater than 0 and less than or equal to 4096.")
```
Comment (xfoggi) [inline src/RobotFrameworkAI/modules/Module.py:108]: 4096 - this is true only for certain models. So if we want to write the actual number, we should have a dictionary/list somewhere with the model = max_tokens

**Pull request merged**